shuffle/cli/src/deploy.rs

```diff
@@ -1,79 +1,110 @@
+fn build_move_packages(project_path: &Path) -> Result<CompiledPackage> {
+    println!("Building Examples...");
+    let pkgdir = project_path.join(MESSAGE_EXAMPLE_PATH);
+    let config = move_package::BuildConfig {
+        dev_mode: true,
+        test_mode: false,
+        generate_docs: false,
+        generate_abis: true,
+    };
 
-pub fn handle(project_dir: PathBuf, account_key_path: PathBuf) -> Result<()> {
-    let config_path = project_dir.join("nodeconfig/0").join("node.yaml");
+    config.compile_package(pkgdir.as_path(), &mut std::io::stdout())
+}
+
+fn publish_packages_as_transaction(project_path: &Path, account_key_path: &Path) -> Result<()> {
+    let config_path = project_path.join("nodeconfig/0").join("node.yaml");
     let config = NodeConfig::load(&config_path)
         .with_context(|| format!("Failed to load NodeConfig from file: {:?}", config_path))?;
     let new_account_key = load_key(account_key_path);
     let json_rpc_url = format!("http://0.0.0.0:{}", config.json_rpc.address.port());
     let factory = TransactionFactory::new(ChainId::test());
-    println!("Connecting to {}...", json_rpc_url);
+    println!("Connecting to {}", json_rpc_url);
 
     let client = BlockingClient::new(json_rpc_url);
 
-    let mut new_account = LocalAccount::new(
-        AuthenticationKey::ed25519(&new_account_key.public_key()).derived_address(),
-        new_account_key,
-        0,
+    println!("Using Public Key {}", &new_account_key.public_key());
+    let derived_address =
+        AuthenticationKey::ed25519(&new_account_key.public_key()).derived_address();
+    println!(
+        "Sending txn from address {}",
+        derived_address.to_hex_literal()
     );
-    // ================= Send a module transaction ========================
-    print!("Add a module to user account...");
-
-    // Get the path to the Move stdlib sources
-    let move_stdlib_dir = move_stdlib::move_stdlib_modules_full_path();
-    let diem_framework_dir = diem_framework::diem_core_modules_full_path();
-    let module_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR"))
-        .join("..")
-        .join(project_dir.clone())
-        .join(project_dir)
-        .join("sources/SampleUserModule.move")
-        .canonicalize()?;
-    let copied_module_path =
-        copy_file_with_sender_address(&module_path, new_account.address()).unwrap();
-    let unwrapped_module_path = copied_module_path.to_str().unwrap();
-    let compiled_module = compile_program(
-        unwrapped_module_path,
-        &[move_stdlib_dir.as_str(), diem_framework_dir.as_str()],
-    )?;
 
-    let publish_txn = new_account.sign_with_transaction_builder(
-        factory.payload(TransactionPayload::Module(Module::new(compiled_module))),
-    );
+    // ================= Send a module transaction ========================
 
-    send(&client, publish_txn)?;
+    let mut new_account = LocalAccount::new(derived_address, new_account_key, 0);
```

Comment:
Does this only work assuming that the new_account has already been made an is in the BC?

Reply:
This command assumes that the new account was already made via `shuffle account`, or it errors out with account not found. Not sure what you mean by BC here?

We're reusing what was there before: 
https://github.com/diem/diem/blob/6d265b6bb7798b5e6b61487adc89b5fc3ec57f60/shuffle/cli/src/deploy.rs#L44-L48